torch/_functorch/aot_autograd.py

```diff
@@ -481,7 +483,7 @@ def __post_init__(self):
         self.aliased_out_indices = aliased_out_indices
         self.num_outputs = len(self.output_info)
         self.num_outputs_non_aliased = len(
-            [x for x in self.output_info if x.output_type == OutputType.non_alias]
+            [x for x in self.output_info if x.output_type in [OutputType.non_alias, OutputType.unsafe_view_alias]]
```

Comment:
Is it worth updating the name of this field? "num_outputs_non_aliased" now that it also counts outputs that may be aliases?

Reply:
My thought here is that for all intents and purposes w.r.t. AOTAutograd, these tensors are non-aliased. I would have liked to not need a new OutputType, and just use `non_alias` for these tensors - but I needed some way of telling the joint code in AOTAutograd to insert an unsafe_view() later (I could have added a separate metadata, but an extra OutputType felt easier).

Let me know if that doesn't seem reasonable to you though!

fwiw, Alban also pointed out another option - don't insert unsafe_view here in AOTAutograd, and instead require the backend compiler to run ADInplaceOrView keys inside of its compiled kernel (that's what happens by default today, since inductor is just generated as_strided()). I figured I would do this simpler thing first that doesn't involve changing inductor code, and leave that for later work.